### PR TITLE
Ensures positive x,y coordinates at graph rendering time. Fixes #184

### LINKED ISSUE #184: Nodes can become unreachable in graph UI
You you drag a table or component off the canvas in the graph UI, it will be unreachable.

#### Steps to reproduce:
1. Open new job
2. Add a table or component to the canvas
3. Make sure window is scroll all the way to the left.
3. Drag table or component off the left side of the main screen.
4. Release.

#### Actual result
Component is now unreachable

#### Expected result
Component should be somehow recoverable, either by refusing to be dropped outside the visible canvas, or by the canvas view extending to fit the new position.

### PR DESCRIPTION
**Pull request by kaspersorensen**
My proposed fix for #184.

@LosD since you reported it, maybe you wanna review?

### CODE REVIEW

File: desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphLayoutTransformer.java [hunk 1]
```diff
@@ -188,6 +188,13 @@ private Point createPoint(final Object vertex, int xIndex, boolean onlyIfCoordin
         if (vertex instanceof Table) {
             JobGraphMetadata.setPointForTable(_analysisJobBuilder, (Table) vertex, point.x, point.y);
         }
+        
+        if (point.x < 0 || point.y < 0) {
+            // apply max(offset, value) to avoid location outside of canvas
+            final int xValue = Math.max(X_OFFSET, point.x);
+            final int yValue = Math.max(Y_OFFSET, point.y);
+            point = new Point(xValue, yValue);
+        }
```
Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphLayoutTransformer.java]: Is this enough? What about the other end of the canvas?

File: desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphMouseListener.java [hunk 2]
```diff
@@ -294,6 +294,10 @@ public void graphReleased(Object v, MouseEvent me) {
                 JobGraphMetadata.setPointForTable(_graphContext.getAnalysisJobBuilder(), (Table) vertex, x, y);
             }
         }
+        
+        if (selectedObjects.length > 0) {
+            _graphContext.getJobGraph().refresh();
+        }
```
Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphMouseListener.java]: I guess we didn't refresh it as often as you thought? :)

File: desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphLayoutTransformer.java (hunk 1 above)
Comment (kaspersorensen) [inline desktop/ui/src/main/java/org/datacleaner/widgets/visualization/JobGraphLayoutTransformer.java]: It only takes into account negative coordinates. I guess the positive limits should not be there, but we need to fix #146 then ... hmm I guess I can add a limit to that for now also.

**Comment by LosD** [comment]
Scope changed a little bit with the addition of the auto-sizing, but looks good!

Let's merge if/when Travis is happy.

**Comment by kaspersorensen** [comment]
OK, travis is happy - I'll merge.

**Pull request merged**